core/node/proof_data_handler/src/tests.rs

```diff
@@ -0,0 +1,247 @@
+//    matches the file from the object store
+#[tokio::test]
+async fn request_tee_proof_inputs() {
+    // prepare a sample mocked TEE verifier input
+
+    let batch_number = L1BatchNumber::from(1);
+    let tvi = TeeVerifierInput::new(
+        PrepareBasicCircuitsJob::new(0),
+        vec![],
+        L1BatchEnv {
+            previous_batch_hash: Some(H256([1; 32])),
+            number: batch_number,
+            timestamp: 0,
+            fee_input: Default::default(),
+            fee_account: Default::default(),
+            enforced_base_fee: None,
+            first_l2_block: L2BlockEnv {
+                number: 0,
+                timestamp: 0,
+                prev_block_hash: H256([1; 32]),
+                max_virtual_blocks_to_create: 0,
+            },
+        },
+        SystemEnv {
+            zk_porter_available: false,
+            version: Default::default(),
+            base_system_smart_contracts: BaseSystemContracts {
+                bootloader: SystemContractCode {
+                    code: vec![U256([1; 4])],
+                    hash: H256([1; 32]),
+                },
+                default_aa: SystemContractCode {
+                    code: vec![U256([1; 4])],
+                    hash: H256([1; 32]),
+                },
+            },
+            bootloader_gas_limit: 0,
+            execution_mode: TxExecutionMode::VerifyExecute,
+            default_validation_computational_gas_limit: 0,
+            chain_id: Default::default(),
+        },
+        vec![(H256([1; 32]), vec![0, 1, 2, 3, 4])],
+    );
+
+    // populate mocked object store with a single batch blob
+
+    let blob_store = ObjectStoreFactory::mock().create_store().await;
+    let object_path = blob_store.put(batch_number, &tvi).await.unwrap();
+
+    // get connection to the SQL db and mock the status of the TEE proof generation
+
+    let db_conn_pool = ConnectionPool::test_pool().await;
+    mock_tee_batch_status(db_conn_pool.clone(), batch_number, &object_path).await;
+
+    // test the /tee_proof_inputs endpoint; it should return the batch from the object store
+
+    let app = create_proof_processing_router(
+        blob_store,
+        db_conn_pool,
+        ProofDataHandlerConfig {
+            http_port: 1337,
+            proof_generation_timeout_in_secs: 10,
+        },
+        L1BatchCommitmentMode::Rollup,
+    );
+    let req_body = Body::from(serde_json::to_vec(&json!({})).unwrap());
+    let response = app
+        .oneshot(
+            Request::builder()
+                .method(Method::POST)
+                .uri("/tee_proof_inputs")
+                .header(http::header::CONTENT_TYPE, mime::APPLICATION_JSON.as_ref())
+                .body(req_body)
+                .unwrap(),
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), StatusCode::OK);
+
+    let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
```

Comment:
This function [seems to be deprecated](https://docs.rs/hyper/0.14.29/src/hyper/body/to_bytes.rs.html#50), and it suggests to use `.collect().await?.to_bytes()` instead. Probably would also make it possible to avoid direct dependency.

Reply:
How did you know? Was it recommended by a linter that you are using? If so, can you recommend one?

Reply:
Nope, I just _really_ do not like adding new top-level dependencies, so I went to the source to check whether the implementation is trivial and can be vendored instead 😅

Reply:
This turned out to be tricky to resolve ([here](https://github.com/matter-labs/zksync-era/pull/1993/commits/581efcdac4fa1eea7dddc5ef231f385cfb17d258) is my attempt to fix this).

The problem boils down to the fact that we are using an old version of the `http` crate. Is there any particular reason for this?
https://github.com/matter-labs/zksync-era/blob/bad5a6c0ec2e166235418a2796b6ccf6f8b3b05f/Cargo.toml#L118
That is why I end up with the following error:
```
error[E0599]: `http_body::combinators::box_body::UnsyncBoxBody<axum::body::Bytes, axum::Error>` is not an iterator
   --> core/node/proof_data_handler/src/tests.rs:109:37
    |
109 |     let body = response.into_body().collect().await.unwrap().to_bytes();
    |                                     ^^^^^^^ `http_body::combinators::box_body::UnsyncBoxBody<axum::body::Bytes, axum::Error>` is not an iterator
    |
   ::: /home/patryk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/http-body-0.4.5/src/combinators/box_body.rs:15:1
    |
15  | pub struct UnsyncBoxBody<D, E> {
    | ------------------------------ doesn't satisfy `_: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `http_body::combinators::box_body::UnsyncBoxBody<axum::body::Bytes, axum::Error>: Iterator`
            which is required by `&mut http_body::combinators::box_body::UnsyncBoxBody<axum::body::Bytes, axum::Error>: Iterator`

For more information about this error, try `rustc --explain E0599`.
```

So, we need to either leave it as-is, using a deprecated function, or upgrade the `http` crate. I'm not sure if there are any other alternatives.

Reply:
OK, here's how to solve this:

1. You added `hyper 1.3.1` to deps, while we transitively depend on `0.14.29` (can be checked in `Cargo.lock`) => Change version in `Cargo.toml`.
2. The method `collect` comes from `HttpBody` trait, not from `Iterator`, as linked above => import said trait via `use hyper::body::HttpBody`.
3. The method was added to `http-body` only in 0.4.6, while we currently depend on 0.4.5 => granularly update via `cargo update http-body --precise 0.4.6`.
4. The code I've copy-pasted from suggestion used `?` to propagate error, while your test doesn't return `Result` => change to `unwrap`.

After doing this I've managed to get your code to compile.

Reply:
That was really helpful! Thank you! [Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/6870aa49f858c830874e8c53e1a2b3488dc3a38b). ✅